### ISSUE TITLE
BUG: Trouble with TreeExplainer, xgboost version, shap version 0.43.0, xgboost version 2.0.2 

### ISSUE THREAD

**Issue by sdcharle** [shap/shap#3413]
### Issue Description

Weird error about 'TypeError: Booster.predict() got an unexpected keyword argument 'ntree_limit''. It looks like the function to predict using DMatrix was deprecated in the latest XGBoost? For now I'll roll back to xgboost 1.9, unless somebody knowns something I don't...

### Minimal Reproducible Example

```python
import shap

explainer = shap.TreeExplainer(model)
shap_values = explainer(dmat_test)

# (dmat_test is of type: xgb.DMatrix)
```

### Traceback

```shell
I get:

TypeError: Booster.predict() got an unexpected keyword argument 'ntree_limit'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File <command-3351324934233699>, line 4
      2 import shap
      3 explainer = shap.TreeExplainer(model)
----> 4 shap_values = explainer(dmat_test)
      5 shap.plots.beeswarm(shap_values)
      7 # what do we have?

File /databricks/python/lib/python3.10/site-packages/shap/explainers/_tree.py:217, in Tree.__call__(self, X, y, interactions, check_additivity)
    214     feature_names = getattr(self, "data_feature_names", None)
    216 if not interactions:
--> 217     v = self.shap_values(X, y=y, from_call=True, check_additivity=check_additivity, approximate=self.approximate)
    218     if type(v) is list:
    219         v = np.stack(v, axis=-1) # put outputs at the end of the
```

### Expected Behavior

Builds the explainer!

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.43.0

**Comment by connortann** [comment]
Thanks for the bug report. Please can you edit your example to be fully self contained and reproducible?

**Comment by bfarrell44** [comment]
Hey there, 

I'm also encountering the same error:      TypeError: predict() got an unexpected keyword argument 'ntree_limit'

I'm using xgboost version 2.0.3 and shap version 0.41.0.

**Comment by connortann** [comment]
> TypeError: predict() got an unexpected keyword argument 'ntree_limit'

This was fixed by #2987 . I would suggest upgrading to the latest release.

For the original issue, we still don't have a complete reproducible example so we can't investigate further. If you still encounter this bug in recent versions of shap, please feel free to open a new issue and please include a minimal example and the full traceback.
